hnn_core/batch_simulate.py

```diff
@@ -84,12 +117,30 @@ def run(self, param_grid, return_output=True, combinations=True, n_jobs=1):
         """
         param_combinations = self._generate_param_combinations(
             param_grid, combinations)
-        results = self.simulate_batch(param_combinations, n_jobs=n_jobs)
+        total_sims = len(param_combinations)
+        num_sims_per_batch = max(total_sims // self.batch_size, 1)
+        batch_size = min(self.batch_size, total_sims)
+
+        results = []
```

Comment:
This may still cause memory problems when you run too many simulations, can you write an option like `clear_cache` that prevents this list from being updates if set to true? (default should be false)